tests/security/check_kernel_config/dm_crypt.pm

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: dm crypt -> add flags to optionally bypass kcryptd
+#          workqueues, the options are 'no-read-workqueue' and
+#          'no-write-workqueue'
+#
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#88873, tc#1768663
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use power_action_utils 'power_action';
+use Utils::Backends 'is_pvm';
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # Make sure the code changes are there
+    assert_script_run("rpm -q kernel-default --changelog | grep 'dm crypt' | grep 'kcryptd workqueues'");
+
+    # Simulate a ram device
+    assert_script_run("modprobe brd rd_nr=1 rd_size=512000");
+
+    # Create dm-crypt devices upon the ram device with different flags
+    my $inline_r_dev = 'eram0-inline-read';
+    my $inline_w_dev = 'eram0-inline-write';
+    assert_script_run("echo '0 1024000 crypt capi:ecb(cipher_null) - 0 /dev/ram0 0 1 no_write_workqueue' | sudo dmsetup create $inline_w_dev");
+    assert_script_run("echo '0 1024000 crypt capi:ecb(cipher_null) - 0 /dev/ram0 0 1 no_read_workqueue' | sudo dmsetup create $inline_r_dev");
+
+    # Check the flags are set correctly
+    assert_script_run("dmsetup table /dev/mapper/$inline_w_dev | grep no_write_workqueue");
+    assert_script_run("dmsetup table /dev/mapper/$inline_r_dev | grep no_read_workqueue");
+
+    # Teardown and release the ram resource
+    power_action("reboot", textmode => 1);
+    reconnect_mgmt_console if is_pvm;
```

Comment:
"reconnect_mgmt_console" already handles backend specifics so you should not need the `if is_pvm` which would be cleaner style as test modules should be backend agnostic.

Reply:
Nice comment, let me do that

Reply:
@okurz 
Seems on s390x platform, "reconnect_mgmt_console" may need some extra efforts there.
https://openqa.suse.de/tests/5508482 (more than 10m)
vs
https://openqa.suse.de/tests/5506944 (about 1m)

Reply:
hm, don't know what caused that runtime difference but I would appreciate if you see any need to adapt that method `reconnect_mgmt_console` to do that within that function and not introduce backend-specifics into the test module itself

Reply:
Thanks Oliver, base our tests, we only need to run this step on powerVM since we need to switch to mgmt console to handle the reboot properly.
And for other platforms, they are qemu/libvirt backends. so we don't need to re-connect the console.

Reply:
However, before just reverting my code, I will try to discuss this case within our team, hopefully we can find a solution :)

Reply:
@okurz 
https://progress.opensuse.org/issues/89041 was filed against the s390x issue. I tested some workaround as well, and the test can passed.

However, to make sure my changes don't impact other tests, I decided to revert my code for now.

Reply:
Sure. I understand. Please keep in mind that the ticket should still be followed up at best by you because there is no magic other team fixing the little differences in behaviour, especially not if only you observed the difference. This might very well be because there is another backend-specific statement somewhere else that would make s390x behave different.

Reply:
Oh, I saw that you already provided something for that ticket yourself. Sorry about that. This is great! Thanks :)

Reply:
Thanks Oliver. I have learnt much form your size on coding style.